src/Language/PureScript/Ide/Externs.hs

```diff
@@ -25,8 +26,8 @@ readExternFile
   -> m P.ExternsFile
 readExternFile fp = do
    externsFile <- liftIO (BS.readFile fp)
-   case decodeStrict externsFile of
-     Nothing ->
+   case Store.decode externsFile of
+     Left _ ->
```

Comment:
I would like to try to come up with a way of parsing the compiler version from an externs file which is independent of the rest of the encoding, so that we can preserve this error message (externs version mismatch). Do you think we could give `ExternsFile` a hand-written `Store` instance which puts the compiler version at the very beginning of the file, and then write a parser for it which just extracts the version and ignores the rest of the file?